test/com/google/enterprise/adaptor/filenet/ObjectMocks.java

```diff
@@ -180,6 +201,81 @@ public static void mockDocumentNotFound(MockObjectStore objectStore,
+      return mockProperty(PropertyFloat64List.class, name, value);
+    } else if (value instanceof Id) {
+      return mockProperty(PropertyId.class, name, value);
+    } else if (value instanceof IdList) {
+      return mockProperty(PropertyIdList.class, name, value);
+    } else if (value instanceof Integer) {
+      return mockProperty(PropertyInteger32.class, name, value);
+    } else if (value instanceof Integer32List) {
+      return mockProperty(PropertyInteger32List.class, name, value);
+    } else if (value instanceof String) {
+      return mockProperty(PropertyString.class, name, value);
+    } else if (value instanceof StringList) {
+      return mockProperty(PropertyStringList.class, name, value);
+    } else if (value == null) {
+      throw new IllegalArgumentException("Property value may not be null.");
+    } else {
+      throw new IllegalArgumentException("Unsupported Property type: "
+          + value.getClass().getName());
+    }
+  }
+
+  /**
+   * Returns a Property with the specified type, name, and value.
+   */
+  public static Property mockProperty(Class<? extends Property>  clazz,
+        String name, Object value) {
+    Property property = createMock(clazz);
+    expect(property.getPropertyName()).andStubReturn(name);
+    if (PropertyBoolean.class.equals(clazz)) {
+      expect(property.getBooleanValue()).andStubReturn((Boolean) value);
+    } else if (PropertyBooleanList.class.equals(clazz)) {
+      expect(property.getBooleanListValue()).andStubReturn((BooleanList) value);
+    } else if (PropertyDateTime.class.equals(clazz)) {
+      expect(property.getDateTimeValue()).andStubReturn((Date) value);
+    } else if (PropertyDateTimeList.class.equals(clazz)) {
+      expect(property.getDateTimeListValue())
+          .andStubReturn((DateTimeList) value);
+    } else if (PropertyFloat64.class.equals(clazz)) {
+      expect(property.getFloat64Value()).andStubReturn((Double) value);
+    } else if (PropertyFloat64List.class.equals(clazz)) {
+      expect(property.getFloat64ListValue()).andStubReturn((Float64List) value);
+    } else if (PropertyId.class.equals(clazz)) {
+      expect(property.getIdValue()).andStubReturn((Id) value);
+    } else if (PropertyIdList.class.equals(clazz)) {
+      expect(property.getIdListValue()).andStubReturn((IdList) value);
+    } else if (PropertyInteger32.class.equals(clazz)) {
+      expect(property.getInteger32Value()).andStubReturn((Integer) value);
+    } else if (PropertyInteger32List.class.equals(clazz)) {
+      expect(property.getInteger32ListValue())
+          .andStubReturn((Integer32List) value);
+    } else if (PropertyString.class.equals(clazz)) {
+      expect(property.getStringValue()).andStubReturn((String) value);
+    } else if (PropertyStringList.class.equals(clazz)) {
+      expect(property.getStringListValue()).andStubReturn((StringList) value);
+    } // Intentionally allow unsupported PropertyTypes to be mocked.
```

Comment:
Yes.  I originally threw IllegalArgumentException (like above), but I needed to test an unsupported type.